shared/src/sys/tokenamount_ser.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright 2019-2022 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use std::borrow::Cow;
+
+use num_bigint::{BigInt, Sign};
+use serde::{Deserialize, Serialize};
+
+use crate::sys::TokenAmount;
+
+/// Wrapper for serializing token amounts to match filecoin spec. Serializes as bytes.
+#[derive(Serialize)]
+#[serde(transparent)]
+pub struct TokenAmountSer<'a>(#[serde(with = "self")] pub &'a TokenAmount);
+
+/// Wrapper for deserializing as TokenAmount from bytes.
+#[derive(Deserialize, Serialize, Clone, Default, PartialEq)]
+#[serde(transparent)]
+pub struct TokenAmountDe(#[serde(with = "self")] pub TokenAmount);
+
+/// Serializes TokenAmount as bytes following Filecoin spec.
+pub fn serialize<S>(token: &TokenAmount, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: serde::Serializer,
+{
+    // review note: safe to deref? should we have a second impl of from with pointer rcvr?
+    let int = crate::econ::TokenAmount::from(*token);
+    // let int: crate::econ::TokenAmount = token.into();
+    let (sign, mut bz) = int.to_bytes_be();
```

Comment:
I'd actually implement this manually:

1. If zero, return an empty buffer.
2. If non-zero, write a single byte (positive), then `int.to_be_bytes()` (after removing the leading zero bytes).